### DISCUSSION TITLE
Acessing hydrated atom value in selectAtom

### DISCUSSION THREAD

**Discussion by HonzaKirchner** [pmndrs/jotai#1888] [Q&A]
Hey,
I started using `useHydrateAtoms` on one of my project and I am struggling to make it work together with `selectAtom`. The problem is that the selector of given selectAtom doesn't have access to the value of the base atom I set in `useHydrateAtoms`.
Here is an [example minimal app](https://codesandbox.io/s/jotai-playground-forked-viye2o?file=/src/App.jsx) that keeps failing because the selector doesn't see initialized value.

Is there any way how to achieve this and get the initialized value in any other way than using regular `useAtom`?

**Comment by dai-shi** [comment]
Please check the signature of `selectAtom`. I think you misunderstood the usage.
https://github.com/pmndrs/jotai/blob/385124fb27e8d441cb3dd75baf399433d27ed82a/src/vanilla/utils/selectAtom.ts#L24-L28

Here's a possible fix: https://codesandbox.io/s/jotai-playground-forked-5ikv8i?file=/src/App.jsx
